ephios/plugins/basesignup/signup/section_based.py

```diff
@@ -233,16 +234,36 @@ def get_configuration_fields(self):
             },
         }
 
-    def get_participant_count_bounds(self):
-        return (
-            sum(section.get("min_count") or 0 for section in self.configuration.sections),
-            sum(
-                section.get("max_count") or section.get("min_count")
-                for section in self.configuration.sections
-            )
-            if all(section.get("max_count") for section in self.configuration.sections)
-            else None,
+    def get_signup_stats(self):
+        from ephios.core.signup.methods import SignupStats
+
+        participations = self.shift.participations.all()
+        requested_count = sum(
+            p.state == AbstractParticipation.States.REQUESTED for p in participations
+        )
+        signup_stats = SignupStats(requested_count, 0, None, 0)
+        section_counter = Counter(
+            participations.values_list("data__dispatched_section_uuid", flat=True)
         )
```

Comment:
Iiterating over participations to get requested_count is fine, because it *is* using the prefeteched queryset.
Calling `values_list` gets you a *new* queryset, which you then evaluate, which hits the DB.

To get the counter, access the uuids manually: 

```suggestion
        participations = list(self.shift.participations.all())
        requested_count = sum(
            p.state == AbstractParticipation.States.REQUESTED for p in participations
        )
        signup_stats = SignupStats(requested_count, 0, None, 0)
        section_counter = Counter(p.data.get("dispatched_section_uuid) for p in participations
        )
```

Beware that the uuid might be for a section that does not exist anymore. What happens then?

Reply:
@jeriox 

about unassigned confirmed participations, do we just subtract these from free & missing on the aggregated stats?

Reply:
as far as I see it, we don't count them and thats fine because the state is not allowed anyways and the planner needs to change it the next time the disposition is saved